### ISSUE TITLE
AI Improvement: Stay At Range

### ISSUE THREAD

**Issue by askebv** [Heishe/idleraiders-second-run#502]
Maybe it could be a Stay At Range upgrade, but it would be nice if priests also stayed at range from the enemies. Basically they should try to stay at a similar range of archers, away from their target, but limited to not be out of healing range of the last target they healed. This should prevent them from running too much around.
Basically find the nearest position within the last healing target circle, excluding the overlap with the circle of the enemy target range.
And even better would be if it would consider the center of all raiders (same as F hotkey finds) and prefer a spot within a small radius of that, if it was within that range. This could be an upgrade in itself though, e.g. "Centralized Healing" or "Nanny Healers" or whatever.

On another note, I don't know if it's fair that raiders can kite bosses. In Frozen Coast, my fire mage kited an Ox from the right edge to the left edge (entrance), and since the archer had gotten himself to the other side of the boss, he kited him the other way - and managed to take him down just before hitting the right edge of the map again.
It is exciting to watch though. And unless boss speed scales with Increased Movement Speed AW, there's not much to do about it, even if we wanted to.

**Comment by TravisGesslein** [comment]
I agree Priest stay at range would be a good idea. Would be a decent chunk of coding work for a small feature though, so it's gonna have low priority until we have important bugs and stuff like that fixed. We will certainly get around to adding new 'ai' upgrades and overhauling or extending old ones, and this is gonna be a good fit for that situation.

Regarding the second thing, yeah Archer's kiting is a pretty awesome but overpowered thing. I think I can deal with it by increasing movement speed of all bosses to a value that is slightly above (upgraded) raider speed. Such that archers can definitely kite for a while, but eventually bosses will catch up to them (even without running into a corner).

**Comment by askebv** [comment]
@Heishe  Regarding kiting, I actually think it's good that bosses are pretty slow by default. It makes them feel more like the giants that they are. If you set all values to be slightly above upgraded speed, it could seem pretty unnatural, e.g. if a boss is accidentally engaged and then almost instantly runs toward you. If you decide to do an anti-kiting fix, I'd prefer either of 2 solutions:

- The boss speed slowly accelerates and is reset every time he starts moving (i.e. stops attacking). You could also do it every time they perform an ability, but then bosses with long attack animations would perhaps becomes too fast (initially).
- If the boss is walking, increase its speed every time it is hit by an ability or attack.
